crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -780,6 +780,15 @@ impl<'a, T: Component> RemovedComponents<'a, T> {
     }
 }
 
+impl<'a, T: Component> IntoIterator for &'a RemovedComponents<'a, T> {
```

Comment:
Technically this doesn't need both lifetimes to be `'a` does it? I think `.iter()` also has an overconstrained lifetime.

Reply:
This does not compile without `&'a` if that's what you're asking.

Reply:
I mean that `iter` returns a `std::iter::Cloned<std::slice::Iter<'_, Entity>>` whose lifetime is linked to the `&self` parameter, but it could return a `std::iter::Cloned<std::slice::Iter<'a, Entity>>`, and by consequence `IntoIterator` could be implemented for `&'a RemovedComponents<'b, T>`, returning a `std::iter::Cloned<std::slice::Iter<'b, Entity>>`.